modules/imgproc/src/warp_affine.simd.hpp

```diff
@@ -0,0 +1,533 @@
+                        if (borderType == BORDER_CONSTANT) {
+                            dstptr[x*3] = bval[0];
+                            dstptr[x*3+1] = bval[1];
+                            dstptr[x*3+2] = bval[2];
+                        }
+                        continue;
+                    }
+
+                    #define FETCH_PIXEL_SCALAR_C3(dy, dx, pxy) \
+                        if ((((unsigned)(ix + dx) < (unsigned)srccols) & ((unsigned)(iy + dy) < (unsigned)srcrows)) != 0) { \
+                            size_t ofs = dy*srcstep + dx*3; \
+                            pxy##r = srcptr[ofs]; \
+                            pxy##g = srcptr[ofs+1]; \
+                            pxy##b = srcptr[ofs+2]; \
+                        } else if (borderType == BORDER_CONSTANT) { \
+                            pxy##r = bval[0]; \
+                            pxy##g = bval[1]; \
+                            pxy##b = bval[2]; \
+                        } else if (borderType == BORDER_TRANSPARENT) { \
+                            pxy##r = dstptr[x*3]; \
+                            pxy##g = dstptr[x*3+1]; \
+                            pxy##b = dstptr[x*3+2]; \
+                        } else { \
+                            int ix_ = borderInterpolate_fast(ix + dx, srccols, borderType_x); \
+                            int iy_ = borderInterpolate_fast(iy + dy, srcrows, borderType_y); \
+                            size_t glob_ofs = iy_*srcstep + ix_*3; \
+                            pxy##r = src[glob_ofs]; \
+                            pxy##g = src[glob_ofs+1]; \
+                            pxy##b = src[glob_ofs+2]; \
+                        }
+                    FETCH_PIXEL_SCALAR_C3(0, 0, p00);
+                    FETCH_PIXEL_SCALAR_C3(0, 1, p01);
+                    FETCH_PIXEL_SCALAR_C3(1, 0, p10);
+                    FETCH_PIXEL_SCALAR_C3(1, 1, p11);
+                }
+                float v0r = p00r + sx*(p01r - p00r);
+                float v0g = p00g + sx*(p01g - p00g);
+                float v0b = p00b + sx*(p01b - p00b);
+
+                float v1r = p10r + sx*(p11r - p10r);
+                float v1g = p10g + sx*(p11g - p10g);
+                float v1b = p10b + sx*(p11b - p10b);
+
+                v0r += sy*(v1r - v0r);
+                v0g += sy*(v1g - v0g);
+                v0b += sy*(v1b - v0b);
+
+                dstptr[x*3] = (uint8_t)(v0r + 0.5f);
+                dstptr[x*3+1] = (uint8_t)(v0g + 0.5f);
+                dstptr[x*3+2] = (uint8_t)(v0b + 0.5f);
+            }
+        }
+    }
+
+private:
+    Mat *output;
+    const Mat *input;
+    int borderType;
+    std::array<float, 6> M;
+    std::array<uint8_t, 4> bval;
+
+    int borderType_x;
+    int borderType_y;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+    static constexpr int vlanes_32{VTraits<v_float32>::max_nlanes};
+    static constexpr int vlanes_16{VTraits<v_int16>::max_nlanes};
```

Comment:
Looks like we cannot use `max_nlanes` for rvv since it is the theoretical upper limit. Lane size is known in runtime.

cc @vpisarev